parachain/pallets/basic-channel/merkle-proof/src/lib.rs

```diff
@@ -0,0 +1,791 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021-2022 Parity Technologies (UK) Ltd.
```

Comment:
I think we need to attribute this code in a `NOTICE` file, as per https://opensource.stackexchange.com/questions/7872/how-to-properly-assemble-notice-file-for-new-software-under-apache-license-2-0. Can you check?

Reply:
It sounds like a NOTICE file is only required for Apache Software Foundation projects:

> If your project is not an Apache Software Foundation project, but just happens to be Apache 2 licensed, you're not required to even keep a NOTICE file.

But the next sentence is:

> The NOTICE file is where you would mention if parts of the code have a different copyright...

This applies here, since our parachain is already under Apache but the copyright belongs to parity for this file.

I'll check in with Parity to see what the acceptable options are.

Reply:
Cool, I guess we’re covered then. The copyright and license text in the header of the file should be sufficient